theme/static/js/hs-vue/left-header-app.js

```diff
@@ -122,6 +122,14 @@ Vue.component('add-author-modal', {
             },
         }
     },
+    mounted(){
+        let that = this;
+        $('.modal-body').click(function(e){
+            if($(e.target).attr('class') === "remove"){
```

Comment:
it actually isn't possible in this case. I believe because the autocomplete-light form (or perhaps Django modelform) is stopping propagation on .clicks internal to the `#user-wrapper`). I will push a non-functional commit that demonstrates this.

I guess I'm still not 100% sure why these vanilla JS or jquery listeners registered on the `mounted()` lifecycle hook are preferable to just creating new method on the Vue template and registering and @click listener (as in my [first commit for this issue](https://github.com/hydroshare/hydroshare/pull/4490/commits/6cf3ab676f259deb9c55e407afa9a162fa161816)) but I trust that there's a good reason I just don't understand because still learning Vue ;)

Reply:
The idea is to bind the click event to the 'remove' button only. 

Having a click event bound to the modal body with logic inside to check if the button was clicked is inefficient, because the event would be triggered when clicking anywhere in the modal body instead of just the button. Furthermore, creates constraints like avoiding having another button there with the 'remove' class since it would also pass the conditional. This makes the code less maintainable.

Reply:
Thanks for the explanation, makes perfect sense!
Looks like, either way we are constrained:
1. vue listener bound directly on the template: we have to register the @click on the `modal-body` and check the class of what was clicked which is inefficient because as you said, it runs the check every time someone clicks in the modal.
2. listener registered in `mounted()`: we have the problem that click propagation is stopped so we have to use a general selector `.modal-body` or have to figure out how to enable propagation--both of which seem like a bad idea.

Is there another option we aren't thinking of?